src/panels/lovelace/editor/section-editor/hui-dialog-edit-section.ts

```diff
@@ -0,0 +1,319 @@
+import { ActionDetail } from "@material/mwc-list";
+import { mdiCheck, mdiClose, mdiDotsVertical } from "@mdi/js";
+import "@polymer/paper-tabs/paper-tab";
+import "@polymer/paper-tabs/paper-tabs";
+import {
+  CSSResultGroup,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+  css,
+  html,
+  nothing,
+} from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { stopPropagation } from "../../../../common/dom/stop_propagation";
+import "../../../../components/ha-button";
+import "../../../../components/ha-button-menu";
+import "../../../../components/ha-circular-progress";
+import "../../../../components/ha-dialog";
+import "../../../../components/ha-dialog-header";
+import "../../../../components/ha-icon-button";
+import "../../../../components/ha-list-item";
+import "../../../../components/ha-yaml-editor";
+import type { HaYamlEditor } from "../../../../components/ha-yaml-editor";
+import { LovelaceSectionRawConfig } from "../../../../data/lovelace/config/section";
+import type { HassDialog } from "../../../../dialogs/make-dialog-manager";
+import { haStyleDialog } from "../../../../resources/styles";
+import type { HomeAssistant } from "../../../../types";
+import {
+  findLovelaceContainer,
+  updateLovelaceContainer,
+} from "../lovelace-path";
+import "./hui-section-settings-editor";
+import "./hui-section-visibility-editor";
+import type { EditSectionDialogParams } from "./show-edit-section-dialog";
+
+const TABS = ["tab-settings", "tab-visibility"] as const;
+
+@customElement("hui-dialog-edit-section")
+export class HuiDialogEditSection
+  extends LitElement
+  implements HassDialog<EditSectionDialogParams>
+{
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: EditSectionDialogParams;
+
+  @state() private _config?: LovelaceSectionRawConfig;
+
+  @state() private _yamlMode = false;
+
+  @state() private _curTab: (typeof TABS)[number] = TABS[0];
+
+  @query("ha-yaml-editor") private _editor?: HaYamlEditor;
+
+  protected updated(changedProperties: PropertyValues) {
+    if (this._yamlMode && changedProperties.has("_yamlMode")) {
+      const viewConfig = {
+        ...this._config,
+      };
+      this._editor?.setValue(viewConfig);
+    }
+  }
+
+  public async showDialog(params: EditSectionDialogParams): Promise<void> {
+    this._params = params;
+
+    this._config = findLovelaceContainer(this._params.lovelaceConfig, [
+      this._params.viewIndex,
+      this._params.sectionIndex,
+    ]) as LovelaceSectionRawConfig;
```

Comment:
For another day, but I feel like we should be able to teach TS that passing an array with 2 numbers to `findLovelaceContainer` always returns `LovelaceSectionRawConfig`